duckdb-vortex/src/vortex_extension.cpp

```diff
@@ -233,27 +316,37 @@ void VortexExtension::Load(DuckDB &db) {
 	DatabaseInstance &instance = *db.instance;
 	TableFunction vortex_func("read_vortex", {LogicalType::VARCHAR}, VortexScanFunction, VortexBind);
 
-	vortex_func.init_local = [](ExecutionContext &context, TableFunctionInitInput &input,
-	                            GlobalTableFunctionState *global_state) -> unique_ptr<LocalTableFunctionState> {
-		auto state = make_uniq<VortexScanLocalState>();
-
-		return state;
-	};
-
 	vortex_func.init_global = [](ClientContext &context,
 	                             TableFunctionInitInput &input) -> unique_ptr<GlobalTableFunctionState> {
+		auto bind = input.bind_data->Cast<VortexBindData>();
 		auto state = make_uniq<VortexScanGlobalState>();
 
 		state->filter = input.filters;
-
-		state->projection_ids = vector<column_t>();
 		state->projection_ids.reserve(input.projection_ids.size());
 		for (auto proj_id : input.projection_ids) {
 			state->projection_ids.push_back(input.column_ids[proj_id]);
 		}
 
 		state->column_ids = input.column_ids;
 
+		// TODO(joe): do this expansion gradually in the scan to avoid a slower start.
+		state->expanded_files = bind.file_list->GetAllFiles();
+
+		// Can ignore mutex since no other threads are running now.
+		state->file_slots[0]->array_stream = OpenArrayStream(bind, *state, bind.initial_file);
+		state->next_file = 1;
+
+		return std::move(state);
+	};
+
+	vortex_func.init_local = [](ExecutionContext &context, TableFunctionInitInput &input,
+	                            GlobalTableFunctionState *global_state) -> unique_ptr<LocalTableFunctionState> {
+		auto &v_global_state = global_state->Cast<VortexScanGlobalState>();
+
+		auto thread_id = v_global_state.thread_id_counter.fetch_add(1);
```

Comment:
`fetch_add` returns the previous, not the new value. But given that you compare against `<` this is correct.

Reply:
I wanted the old one, what do you want me to compare?

Reply:
As in this is good as is.